File: LeetCode_Solutions/LeetCode_Solutions/FizzBuzz_412.m
//
//  FizzBuzz_412.m
//  LeetCode_Solutions
//
//  Created by 来自遥远星系的核心巡洋舰 on 16/10/28.
//  Copyright © 2016年 KnightJoker. All rights reserved.
//




#import "FizzBuzz_412.h"

@implementation FizzBuzz_412

- (NSMutableArray *)fizzBuzz:(NSInteger)n {


    NSMutableArray *array = [[NSMutableArray alloc] init];
    for (int i = 1; i < n + 1; i++) {
        if (i % 3 == 0 && i % 5 != 0) {
            [array addObject:@"Fizz"];
        } else if (i % 3 != 0 && i % 5 == 0) {
            [array addObject:@"Buzz"];
        } else if (i % 3 == 0 && i % 5 == 0) {
            [array addObject:@"FizzBuzz"];
        } else {
            [array addObject:@(i)];
        }
    }
    
    return array;
}

@end

File: LeetCode_Swift/LeetCode_Swift/FizzBuzz_412.swift
//
//  FizzBuzz_412.swift
//  LeetCode_Swift
//
//  Created by Huni on 16/08/2017.
//  Copyright © 2017 KnighhtJoker. All rights reserved.
//

import Foundation

public class FizzBuzz_412 {

    func fizzBuzz(_ n: Int) -> [String] {
        var stringArray = [String]()
        var count = 1
        while count <= n {
            if count % 3 == 0 && count % 5 != 0{
                stringArray.append("Fizz")
            } else if count % 5 == 0 && count % 3 != 0{
                stringArray.append("Buzz")
            } else if count % 3 == 0 && count % 5 == 0 {
                stringArray.append("FizzBuzz")
            } else {
                stringArray.append(String(count))
            }
            count += 1
        }
        return stringArray
    }
}
